common/alog.h

```diff
@@ -652,16 +601,12 @@ struct __limit_first_n_every_t {
     }
 };
 
-#define __LOG_WITH_LIMIT(type, ...)                                  \
-    [&] {                                                            \
-        static type limiter;                                         \
-        auto __logstat__ = __VA_ARGS__;                              \
-        __logstat__.done = limiter();                                \
-        return __log_append_tail(std::move(__logstat__),             \
-                                 [](STFMTLogBuffer buffer) {         \
-                                     limiter.append_tail(buffer);    \
-                                 });                                 \
-    }()
+#define __LOG_WITH_LIMIT(type, ...) do {    \
+    static type limiter;                    \
+    if (!limiter()) {                       \
+        __VA_ARGS__;                        \
+    }                                       \
+} while(0)
```

Comment:
LOG_WITH_LIMITS without tail is confusing.
Since currently __LOG__ is not a value any more, append tail after LogBuffer build is quite hard.
Change the usage of those `LOG_EVERY_T` `LOG_EVERY_N` might be better.

Such as `LOG_EVERY_T(10, LOG_INFO(...))` change to `LOG_INFO(LOG_EVERY_T(10), ...)`